pkg/cloud/services/networking/floatingip.go

```diff
@@ -53,6 +56,16 @@ func (s *Service) GetOrCreateFloatingIP(openStackCluster *infrav1.OpenStackClust
 		return nil, err
 	}
 
+	if len(openStackCluster.Spec.Tags) > 0 {
```

Comment:
if possible, can we check add options in create by using this
https://github.com/gophercloud/gophercloud/blob/master/openstack/networking/v2/extensions/layer3/floatingips/requests.go#L73 instead of update?

Reply:
If I get you right, you want to add the tags with the `CreateOpts` instead of an extra update API call?
Unfortunately, this is not possible (see https://github.com/gophercloud/gophercloud/issues/2184).
Further, we are already updating the tags in the [network.go](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/pkg/cloud/services/networking/network.go#L136-L143) ([twice](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/pkg/cloud/services/networking/network.go#L240-L248)) and the [router.go](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/pkg/cloud/services/networking/router.go#L149-L156)

Reply:
ok, make sense for now , thanks~